src/posts/posts.go

```diff
@@ -0,0 +1,49 @@
+package posts
+
+import (
+	"net/url"
+
+	"github.com/gin-gonic/gin"
+)
+
+// AddRoutes ...
+// Adds HTTP Methods for the /posts endpoint
+func AddRoutes(router *gin.Engine) {
+	router.GET("/posts", func(c *gin.Context) {
+
+		// Check queries
+		queries := c.Request.URL.Query()
+
+		if checkIfQueryExist(queries) {
+			c.JSON(200, gin.H{
+				"result": queries,
+			})
+		} else {
+			c.JSON(200, gin.H{
+				"result": []string{"test", "sample", "post"},
+			})
+		}
+	})
+
+	router.GET("/posts/:id", func(c *gin.Context) {
+		id := c.Param("id")
+
+		c.JSON(200, gin.H{
+			"result": id,
+		})
+	})
+
+	router.GET("/posts/:id/like", func(c *gin.Context) {
+		_ = c.Param("id")
+		c.JSON(200, gin.H{
+			"result": 10,
+		})
+	})
+}
+
+func checkIfQueryExist(values url.Values) bool {
```

Comment:
이거 약간 영어 잘못 알려준듯 쿼리들이 있는지 확인하려면
```suggestion
func checkIfQueriesExist(values url.Values) bool {
```

쿼리 하나 있는지 확인하려면
```suggestion
func checkIfQueryExists(values url.Values) bool {
```

영어 잘하는 분이 잘 봐주셨으면 좋겠네요 ㅎㅎ

Reply:
앗....